apps/src/lab2/views/LabViewsRenderer.tsx

```diff
@@ -32,6 +33,12 @@ const LabViewsRenderer: React.FunctionComponent = () => {
   const exemplarSources = useAppSelector(
     state => state.lab.levelProperties?.exemplarSources
   );
+  const isBlocked = useAppSelector(state => state.lab.isBlocked);
+  const permissions = useAppSelector(state => state.lab.permissions);
+  const isProjectValidator = permissions?.includes(
+    PERMISSIONS.PROJECT_VALIDATOR
+  );
```

Comment:
nit/optional: can compute `isProjectValidator` in the selector directly, like `isProjectValidator = useAppSelector(state => state.lab.permissions?.includes(...))`

You could also just have one selector that computes the overall blocked state, something like
```
const hideProject = useAppSelector(state => state.lab.isBlocked && !state.lab.permissions?.includes(...))
```
So the component would only re-render if the overall `hideProject` value changes. Totally optional though, all of this is pretty fast and readable as it is.

Reply:
Oh nice! Just updated to compute `isProjectValidator` directly. I'm going to keep the rest of change as-is for readability. Thanks!